### PR TITLE
Improve Rust Documentation

### LINKED ISSUE #3303: Rust bindings only works with path dependency
Using dependency from crates.io doesn't work.
I can only use the rust bindings if I compile them in the Elektra proejct first, and then use the compiled (replace macros) directory as a path dependency.  

## Steps to Reproduce the Problem

```sh
# Create a new cargo project
cargo new test-elektra-rs
cd test-elektra-rs
# Add the elektra dependency
echo "elektra = \"0.9\"" >> Cargo.toml
# Build the project
cargo build
```

## Expected Result

Build successful

## Actual Result

Error:
```
error[E0425]: cannot find function `keyIsDirectBelow` in crate `elektra_sys`
   -->CARGO_DIR/registry/src/github.com/elektra-0.9.0/src/readable.rs:231:31
    |
231 |         unsafe { elektra_sys::keyIsDirectBelow(other.as_ref(), self.as_ref()) == 1 }
    |                               ^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `keyIsDirectlyBelow`

error[E0425]: cannot find value `KDB_O_NOCASE` in crate `elektra_sys`
  --> CARGO_DIR/registry/src/github.com/elektra-0.9.0/src/keyset.rs:24:43
   |
24 |         const KDB_O_NOCASE = elektra_sys::KDB_O_NOCASE as elektra_sys::option_t;
   |                                           ^^^^^^^^^^^^ help: a constant with a similar name exists: `KDB_O_NONE`

error[E0425]: cannot find value `KDB_O_WITHOWNER` in crate `elektra_sys`
  --> CARGO_DIR/registry/src/github.com/elektra-0.9.0/src/keyset.rs:26:46
   |
26 |         const KDB_O_WITHOWNER = elektra_sys::KDB_O_WITHOWNER as elektra_sys::option_t;
   |                                              ^^^^^^^^^^^^^^^ not found in `elektra_sys`

error[E0425]: cannot find value `KDB_O_NOALL` in crate `elektra_sys`
  --> CARGO_DIR/registry/src/github.com/elektra-0.9.0/src/keyset.rs:28:42
   |
28 |         const KDB_O_NOALL = elektra_sys::KDB_O_NOALL as elektra_sys::option_t;
   |                                          ^^^^^^^^^^^ help: a constant with a similar name exists: `KDB_O_NONE`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0425`.
error: could not compile `elektra`.

To learn more, run the command again with --verbose.
```

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
fixes #3303 

The old `0.9.0` release of the elektra crate on `crates.io` was no longer comptaible with the latest elektra `0.9.1` release. I just published new versions to `crates.io`. I took the opportunity to improve the documentation further.

The Readme is now also included on the [crates.io page](https://crates.io/crates/elektra) so I made some changes for compatibility, such as absolute instead of relative links.

`elektra-sys` also needed a new version, since `0.9.0` was still linking against the single `elektra` shared library.

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [x] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you so much for working on it!

**Comment by PhilippGackstatter** [comment]
jenkins build libelektra please

File: src/bindings/rust/README.md [hunk 1]
```diff
@@ -4,13 +4,28 @@
...
+  <!-- docs.rs docs -->
+  <a href="https://docs.rs/elektra">
+    <img src="https://img.shields.io/badge/docs-latest-blue.svg"
+      alt="docs.rs docs" />
+  </a>
+</div>
+
+_Elektra serves as a universal and secure framework to access configuration parameters in a global, hierarchical key database._
+
+For more information about elektra itself, visit the [website](https://libelektra.org).
```
Comment (mpranj) [inline src/bindings/rust/README.md]:
```suggestion
For more information about Elektra itself, visit the [website](https://libelektra.org).
```

We write it with a capital E, I'm not sure it's properly documented.

File: src/bindings/rust/README.md [hunk 2]
```diff
@@ -30,7 +45,7 @@ With this in place, the bindings should be built when you run `cargo build`.
 
 ### Local Build
 
-To build the bindings explicitly as part of the elektra build process, we add the option `rust` to `-DBINDINGS`. Now [build libelektra](../../../doc/COMPILE.md) and the bindings will be built as part of this process.
+To build the bindings explicitly as part of the elektra build process, we add the option `rust` to `-DBINDINGS`. Now [build libelektra](https://master.libelektra.org/doc/COMPILE.md) and the bindings will be built as part of this process.
```
Comment (mpranj) [inline src/bindings/rust/README.md]:
```suggestion
To build the bindings explicitly as part of the Elektra build process, we add the option `rust` to `-DBINDINGS`. Now [build libelektra](https://master.libelektra.org/doc/COMPILE.md) and the bindings will be built as part of this process.
```

File: src/bindings/rust/README.md (hunk 1 above)
Comment (markus2330) [reply src/bindings/rust/README.md]: It is an "Eigenname", so it is simply grammar that tells us to make a big "E".

**Comment by mpranj** [comment]
jenkins build libelektra please

**Comment by mpranj** [comment]
Thank you so much!

I will assume capitalization did not introduce critical problems and merge it, since Jenkins seems to have problems again. 😢

**Pull request merged**